packages/tasit-action/src/NFT.test.js

```diff
@@ -189,5 +205,103 @@ describe("TasitAction.NFT", () => {
       const balanceAfter = await fullNFT.balanceOf(bob.address);
       expect(balanceAfter.toNumber()).to.equal(1);
     });
+
+    it("should transfer an owned token using safeTransferFrom", async () => {
+      let senderBalance, receiverBalance;
+
+      fullNFT = new NFT(fullNFTAddress, ana);
+
+      senderBalance = await fullNFT.balanceOf(ana.address);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(bob.address);
+      expect(receiverBalance.toNumber()).to.equal(0);
+
+      txSubscription = fullNFT.safeTransferFrom(
+        ana.address,
+        bob.address,
+        tokenId
+      );
+
+      await txSubscription.waitForNonceToUpdate();
+
+      senderBalance = await fullNFT.balanceOf(ana.address);
+      expect(senderBalance.toNumber()).to.equal(0);
+
+      receiverBalance = await fullNFT.balanceOf(bob.address);
+      expect(receiverBalance.toNumber()).to.equal(1);
+    });
+
+    it("shouldn't do a safeTransferFrom to a contract that doesn't implement `onERC721Received` - Contract Subscription", async () => {
+      let senderBalance, receiverBalance;
+      const sender = ana.address;
+      const receiver = simpleStorageWithRemovedAddress;
+      const contractErrorFakeFn = sinon.fake();
+
+      fullNFT = new NFT(fullNFTAddress, ana);
+
+      senderBalance = await fullNFT.balanceOf(sender);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(receiver);
+      expect(receiverBalance.toNumber()).to.equal(0);
+
+      const contractErrorListener = message => {
+        const { error, eventName } = message;
+        expect(error.message).to.equal(
+          "Action with error: VM Exception while processing transaction: revert"
+        );
+        contractErrorFakeFn();
+      };
+
+      fullNFT.on("error", contractErrorListener);
+
+      txSubscription = fullNFT.safeTransferFrom(sender, receiver, tokenId);
+      await txSubscription.waitForNonceToUpdate();
+
+      expect(contractErrorFakeFn.called).to.be.true;
+
+      senderBalance = await fullNFT.balanceOf(sender);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(receiver);
+      expect(receiverBalance.toNumber()).to.equal(0);
+    });
+
+    it("shouldn't do a safeTransferFrom to a contract that doesn't implement `onERC721Received` - Action Subscription", async () => {
```

Comment:
It's not clear enough from the name of the test case how this will be different from the test case above ("shouldn't do a safeTransferFrom to a contract that doesn't implement `onERC721Received` - Contract Subscription")

Reply:
```suggestion
    it("should trigger an error if the user is listening for errors for an action and tries safeTransferFrom to a contract without onERC721Received", async () => {
```

Reply:
If this looks too verbose, we could always use the same text for the `it` but put a different `describe` above the two similar `it`'s